### PR TITLE
Switch to debian bullseye base image

### PR DESCRIPTION
**Pull request by francoisferrand**
- Upgrade to debian bullseye baseline
- Use build arg to specify node version once
- Switch to go-lang supervisor

Issue: S3UTILS-73

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,35 +1,38 @@
...
     apt-get install -y --no-install-recommends \
         build-essential \
+        ca-certificates \
         git \
         jq \
         python3 \
-        ssh
+        ssh \
+        wget \
+    && rm -rf /var/lib/apt/lists/*
```
Comment (jbertran) [inline Dockerfile:17]: We can `apt clean` instead - it's possible to to a dry run of that command to check if we're cleaning enough.

**Review by jbertran — COMMENTED** [review]
Re: supervisord, have you tried the documented docker-based approach (https://github.com/ochinchina/supervisord#usage-from-a-docker-container)? It seems less brittle and less error-prone than the .tar way.

File: Dockerfile (hunk 1 above)
Comment (francoisferrand) [reply Dockerfile:17]: according to https://docs.docker.com/develop/develop-images/dockerfile_best-practices/, `apt clean` is supposed to be automatically called... but the exemple does the `rm -rf /var/lib/apt/lists/*` (just like it was before this change, see later in the diff)

so i choose to stick to the old & proven approach 😀

**Comment by francoisferrand** [comment]
> Re: supervisord, have you tried the documented docker-based approach ([ochinchina/supervisord#usage-from-a-docker-container](https://github.com/ochinchina/supervisord#usage-from-a-docker-container))? It seems less brittle and less error-prone than the .tar way.

actually, the images are not tagged, and not even up-to-date.

**Comment by jbertran** [comment]
:man_facepalming:  LGTM then!

**Review by jbertran — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**